lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -123,6 +136,23 @@ def update_source(status)
           logger.availability_check("Failed to update Source id:#{source_id} - #{e.message}", :error)
         end
 
+        def update_source_by_kafka(status)
+          client = ManageIQ::Messaging::Client.open(:protocol => :Kafka, :host => messaging_host, :port => messaging_port)
+          client.publish_message(
+            :service => SERVICE_NAME,
+            :message => EVENT_AVAILABILITY_STATUS,
+            :payload => {
+              :resource_type => "source",
+              :resource_id   => source_id,
+              :status        => status
+            }.to_json
+          )
```

Comment:
I think this should be a method that receives the payload as a parameter, rather than duplicating it 3 times. Maybe refactor this into a `availability_status_message(payload)` method.